fission/src/test/bootstrap/ContextProviders.test.tsx

```diff
@@ -0,0 +1,273 @@
+        unmountToast()
+
+        // Test TooltipControlProvider
+        const mockTooltipMethods = {
+            showTooltip: vi.fn(),
+        }
+        const { unmount: unmountTooltip } = render(
+            <TooltipControlProvider {...mockTooltipMethods}>
+                <TestComponent />
+            </TooltipControlProvider>
+        )
+        expect(screen.getByTestId("test-component")).toBeDefined()
+        unmountTooltip()
+    })
+
+    test("all providers work together in nested structure", () => {
+        const mockModalMethods = {
+            openModal: vi.fn(),
+            closeModal: vi.fn(),
+            activeModalId: null,
+        }
+
+        const mockPanelMethods = {
+            openPanel: vi.fn(),
+            closePanel: vi.fn(),
+            closeAllPanels: vi.fn(),
+        }
+
+        const mockTooltipMethods = {
+            showTooltip: vi.fn(),
+        }
+
+        render(
+            <ThemeProvider initialThemeName="Default" themes={mockThemes} defaultTheme={mockTheme}>
+                <TooltipControlProvider {...mockTooltipMethods}>
+                    <ModalControlProvider {...mockModalMethods}>
+                        <PanelControlProvider {...mockPanelMethods}>
+                            <ToastProvider>
+                                <TestComponent />
+                            </ToastProvider>
+                        </PanelControlProvider>
+                    </ModalControlProvider>
+                </TooltipControlProvider>
+            </ThemeProvider>
+        )
+
+        expect(screen.getByTestId("test-component")).toBeDefined()
+        expect(screen.getByTestId("context-test")).toBeDefined()
+    })
+
+    test("theme provider applies CSS variables correctly", () => {
+        render(
+            <ThemeProvider initialThemeName="Default" themes={mockThemes} defaultTheme={mockTheme}>
+                <TestComponent />
+            </ThemeProvider>
+        )
+
+        // The theme provider should set CSS custom properties on document root
+        // We can't easily test this in JSDOM, but we can verify the component renders
+        expect(screen.getByTestId("test-component")).toBeDefined()
+    })
+
+    test("providers handle edge cases and maintain robustness", () => {
+        // Test providers with missing children
+        expect(() => {
+            render(
+                <ThemeProvider initialThemeName="Default" themes={mockThemes} defaultTheme={mockTheme}>
+                    {null}
+                </ThemeProvider>
+            )
+        }).not.toThrow()
+
+        // Test providers with minimal props
+        expect(() => {
+            render(
+                <ThemeProvider initialThemeName="Default" themes={mockThemes} defaultTheme={mockTheme}>
+                    <TestComponent />
+                </ThemeProvider>
+            )
+        }).not.toThrow()
+
+        // Test with empty theme object
```

Comment:
I don't think this is actually a useful test to have if this isn't a scenario that ever occurs. It is, in my opinion, not useful to test edge cases by just checking for an error if those edge cases won't come up, and you can check if they do by just checking if the main app renders